web/packages/teleport/src/SessionRecordings/view/ProgressBar/ProgressBar.tsx

```diff
@@ -16,23 +16,51 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React, { memo } from 'react';
+import React, { memo, useCallback, useEffect, useRef } from 'react';
 import styled from 'styled-components';
 
 import Flex from 'design/Flex';
 import * as Icons from 'design/Icon';
 import { Layout } from 'design/Icon';
 import { HoverTooltip } from 'design/Tooltip';
+import { useResizeObserver } from 'design/utils/useResizeObserver';
 
 import Slider from './Slider';
 
+interface SliderHandle {
+  _handleResize: () => void;
+}
+
 export default function ProgressBar(props: ProgressBarProps) {
   const Icon = props.isPlaying ? Icons.CirclePause : Icons.CirclePlay;
+  const sliderRef = useRef<SliderHandle>(null);
+
+  const containerRef = useResizeObserver(
+    useCallback(() => {
```

Comment:
Are you sure it's safe to call `useCallback` inside `useResizeObserver` like this?